src/Build.UnitTests/WarningsAsMessagesAndErrors_Tests.cs

```diff
@@ -84,6 +82,45 @@ public void TreatWarningsAsMessagesWhenSpecified()
             VerifyBuildMessageEvent(logger);
         }
 
+        /// <summary>
+        /// https://github.com/Microsoft/msbuild/issues/2667
+        /// </summary>
+        [Fact]
+        public void TreatWarningsAsMessagesWhenBuildingSameProjectMultipleTimes()
```

Comment:
Maybe another test for `WarningsAsErrors`, to be thorough?

Reply:
Done